src/cargo/ops/fix.rs

```diff
@@ -144,15 +144,18 @@ pub fn fix_maybe_exec_rustc() -> CargoResult<bool> {
     // If we didn't actually make any changes then we can immediately exec the
     // new rustc, and otherwise we capture the output to hide it in the scenario
     // that we have to back it all out.
-    if !fixes.original_files.is_empty() {
+    if !fixes.files.is_empty() {
         let mut cmd = Command::new(&rustc);
         args.apply(&mut cmd);
         cmd.arg("--error-format=json");
         let output = cmd.output().context("failed to spawn rustc")?;
 
         if output.status.success() {
-            for message in fixes.messages.drain(..) {
-                message.post()?;
+            for (path, file) in fixes.files.iter() {
```

Comment:
any reason to explicitly call `iter` instead of just `&fixes.files`? (same below)